src/chat.rs

```diff
@@ -3319,6 +3318,8 @@ pub(crate) async fn mark_old_messages_as_noticed(
     context: &Context,
     mut msgs: Vec<ReceivedMsg>,
 ) -> Result<()> {
+    debug_assert!(!msgs.is_empty());
```

Comment:
Why? It's a no-op if `msgs` is empty, moreover, `fetch_new_messages()` calls this function unconditionally